ml4gw/dataloading/in_memory_dataset.py

```diff
@@ -191,7 +191,9 @@ def init_indices(self):
             # indices we'll need rather than having to generate
             # everything.
             idx = [range(self.num_kernels) for _ in range(len(self.X))]
-            idx = zip(range(num_kernels), itertools.product(*idx))
+            idx = zip(
+                range(num_kernels), itertools.product(*idx), strict=False
```

Comment:
Turns out this the only place where we need the strict=False for zip since the itertools.product can return a longer iterator